openraft/src/core/raft_core.rs

```diff
@@ -1080,7 +1080,7 @@ impl<C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> RaftCore<C,
             self.config.clone(),
             self.engine.state.last_log_id(),
             self.engine.state.committed,
-            self.network.connect(target, target_node).await,
+            self.network.connect(target, target_node).await.unwrap(),
```

Comment:
For a failure to create a replication stream, one of the choices is to return a `None` as a placeholder in `LeaderData.nodes`.

Reply:
I don't really get it, do you mean change self.leader_data into a `None`? this will also lead this function's caller to reach an `unreachable!()` arm and crash.

Reply:
> I don't really get it, do you mean to change `self.leader_data` into a `None` in this method? this will also lead the method's caller to reach an `unreachable!()` arm and crash.

Sorry, I mean:
If it creates a network successfully, a ReplicationStream will be inserted into `leader_data.nodes`. If it fails to create a network, just insert a `None` into `leader_data.nodes`.